public/app/features/transformers/FilterByValueTransformer/ValueMatchers/BasicMatcherEditor.test.tsx

```diff
@@ -0,0 +1,53 @@
+import { render, screen, fireEvent, waitFor } from '@testing-library/react';
+
+import { FieldType } from '@grafana/data';
+import { getTemplateSrv } from '@grafana/runtime';
+
+import { basicMatcherEditor } from './BasicMatcherEditor';
+
+jest.mock('@grafana/runtime', () => ({
+  getTemplateSrv: jest.fn(),
+}));
+
+describe('BasicMatcherEditor', () => {
+  it('shows variable suggestions', async () => {
+    // Mock template service variables
+    const mockVariables = [
+      { name: 'var1', label: 'Variable 1', value: '$var1' },
+      { name: 'var2', label: 'Variable 2', value: '$var2' },
+    ];
+
+    (getTemplateSrv as jest.Mock).mockReturnValue({
```

Comment:
Consider using [jest.mocked](https://jestjs.io/docs/mock-function-api#jestmockedsource-options) to get a more type-rich mock